homeassistant/components/zerproc/light.py

```diff
@@ -28,34 +28,24 @@
 
 DISCOVERY_INTERVAL = timedelta(seconds=60)
 
-PARALLEL_UPDATES = 0
 
-
-def connect_lights(lights: List[pyzerproc.Light]) -> List[pyzerproc.Light]:
-    """Attempt to connect to lights, and return the connected lights."""
-    connected = []
-    for light in lights:
-        try:
-            light.connect()
-            connected.append(light)
-        except pyzerproc.ZerprocException:
-            _LOGGER.debug("Unable to connect to '%s'", light.address, exc_info=True)
-
-    return connected
-
-
-def discover_entities(hass: HomeAssistant) -> List[Entity]:
+async def discover_entities(hass: HomeAssistant) -> List[Entity]:
     """Attempt to discover new lights."""
-    lights = pyzerproc.discover()
+    lights = await pyzerproc.discover()
 
     # Filter out already discovered lights
     new_lights = [
         light for light in lights if light.address not in hass.data[DOMAIN]["addresses"]
     ]
 
     entities = []
-    for light in connect_lights(new_lights):
-        # Double-check the light hasn't been added in another thread
+    for light in new_lights:
+        try:
+            await light.connect()
```

Comment:
Can we use `asyncio.gather` to connect concurrently or is there something that requires this to run sequentially? Some APIs don't handle concurrent calls well.

Reply:
TBH I'm not sure how well the underlying library handles concurrency. It's probably fine, since the underlying library uses the BlueZ DBus API which I can expect can handle it, but I was just being conservative with it since it's only happening during discovery.

I'll give it a try, and if it seems to be working fine I'll open a quick follow-up PR.

Reply:
Done, thanks for the review!

https://github.com/home-assistant/core/pull/44376